src/last-state-with-input/lastStateWithInput.js

```diff
@@ -1,17 +1,19 @@
 function getLastStateWithInput(stateName, stateId, lastStateWithInputCurrent) {
+    let nowDate = new Date().toLocaleDateString('pt-BR');
     let lastStateWithInput = {
         name: stateName,
         id: stateId,
+        date: nowDate,
         previousName: null,
-        previousId: null
+        previousId: null,
+        previousDate: null
     };
 
     try {
         lastStateWithInputCurrentParsed = JSON.parse(lastStateWithInputCurrent);
```

Comment:
declare this variable. 

const lastStateWithInputCurrentParsed = JSON.parse(lastStateWithInputCurrent);